test/router-luatest/map_callrw_test.lua

```diff
@@ -647,3 +647,40 @@ g.test_map_callrw_array_encoded_as_map = function(cg)
         end
     end, nil, bids)
 end
+
+--
+-- gh-594: using cdata as bucket_id should be prohibited, as it causes fullscan
+-- of the cluster and subsequent memory leak in the route_map.
+--
+g.test_map_callrw_with_cdata_bucket_id = function(cg)
+    cg.router:exec(function()
+        -- vshard.router._buckets_group is checked for the sake of unit
+        -- testing of the map_callrw() protection.
+        local msg = 'is not a number'
+        local _, err = ivshard.router._buckets_group({1ULL, 2ULL}, 1)
+        ilt.assert_str_contains(err.message, msg)
+        _, err = ivshard.router._buckets_group({1, 2ULL}, 1)
+        ilt.assert_str_contains(err.message, msg)
+        _, err = ivshard.router._buckets_group({1ULL, 2}, 1)
+        ilt.assert_str_contains(err.message, msg)
+
+        -- Map_callrw in general forbids using cdata.
+        _, err = ivshard.router.map_callrw('assert', {'a'},
+            {bucket_ids = {1, 2ULL}})
+        ilt.assert_str_contains(err.message, msg)
+        _, err = ivshard.router.map_callrw('assert', {'a'},
+            {bucket_ids = {[1] = 'a', [2ULL] = 'b'}})
+        ilt.assert_str_contains(err.message, msg)
+        _, err = ivshard.router.map_callrw('assert', {'a'},
+            {bucket_ids = {[1ULL] = 'a', [2ULL] = 'b'}})
+
+        local res
+        -- But there's one case, where it's not possible to forbid that,
```

Comment:
If smb sees the solution for that problem, please, share the ideas)

Reply:
Or you can ban cdata at all because I don't know how you can handle correctly following case:
```
tarantool> {[1LL] = 10, [1LL] = 20, [1LL] = 30}
---
- 1: 20
  1: 30
  1: 10
...
```

Reply:
Yeah, I don't really wanna see cdata at all in bucket_ids. One of the reason, is `map_callrw`, another one - the majority of methods already prohibits cdata as `bucket_id`, we just need to prohibit it for remainings.

>

```
tarantool> {[1LL] = 10, [1LL] = 20, [1LL] = 30}
---
- 1: 20
  1: 30
  1: 10
...
```

This case is already prohibited, there're no problems with prohibiting, when cdata is in key of the table. There're problems with the situation ,when it's a value. The array {10LL, 20LL} is actually {1 = 10LL, 2 = 20LL} and I cannot prohibit that, since passing cdata as values to functions is ok, but user probably means, that the `map_callrw` should be executed on 10th and 20th bucket, not on the first two of them